test/libyul/yulOptimizerTests/commonSwitchCasePrefixMover/sstoreExpression.yul

```diff
@@ -0,0 +1,34 @@
+{
+  function f() -> x {
+    x := calldataload(0)
+    sstore(0, 42)
+  }
+  switch f()
```

Comment:
Perhaps add tests for `switch mload()`, `switch sload` and environment related opcodes like `switch msize()`?